fvm/transactionInvoker.go

```diff
@@ -298,27 +298,31 @@ func (executor *transactionExecutor) normalExecution() (
 		return
 	}
 
-	// log the execution intensities here, so that they do not contain data
-	// from storage limit checks and transaction deduction, because the payer
-	// is not charged for those.
-	executor.logExecutionIntensities()
-
-	executor.txnState.RunWithAllLimitsDisabled(func() {
-		err = executor.deductTransactionFees()
-	})
-	if err != nil {
-		return
-	}
-
 	// Check if all account storage limits are ok
 	//
 	// disable the computation/memory limit checks on storage checks,
 	// so we don't error from computation/memory limits on this part.
-	// We cannot charge the user for this part, since fee deduction already happened.
+	// The storage limit check for the payer assumes that the payers
+	// balance is max_fees lower than the current payers balance,
+	// because the payer will have to pay for this tx.
```

Comment:
Let me try to describe this in a more concise way:

```suggestion
	// The storage limit check is performed for all accounts that were touched during the transaction.
	// The storage capacity of an account depends on its balance and should be higher than the accounts storage used.
	// The payer account is special cased in this check and its balance is considered max_fees lower than 
	// its actual balance, for the purpose of calculating storage capacity, because the payer will have to pay for this tx.
```

@zhangchiqing is this easier to understand whats going on?